client/src/app/shared/user/user.effect.ts

```diff
@@ -5,27 +5,28 @@ import { of } from 'rxjs';
 import { tap, map, exhaustMap, catchError } from 'rxjs/operators';
 import { HttpService } from '../../core/http/http.service';
 import { LoginResult } from '../../login/shared/login-result.model';
-import * as UserAction from './user.action';
+import { UserActions, login, loginSuccess, loginFailure } from './user.action';
 
 @Injectable()
 export class UserEffect {
   @Effect()
   login$ = this.actions$.pipe(
-    ofType(UserAction.UserActionTypes.LOGIN),
-    map((action: UserAction.Login) => action.payload),
+    ofType(login.type),
+    map(action => action.payload),
     exhaustMap(payload => {
       return this.httpService.post<LoginResult>('/login', payload).pipe(
-        map(result => new UserAction.LoginSuccess({ user: result.data })),
-        catchError(error => of(new UserAction.LoginFailure({ error: error.error })))
+        map(result => loginSuccess({ user: result.data })),
+        catchError(error => of(loginFailure({ error: error.error })))
```

Comment:
__[TSLint]__ of is deprecated: use {@link scheduled} instead `scheduled([a, b, c], scheduler)` (deprecation) ([link](https://sider.review/gh/repos/72339652/pulls/67#issue-4029942))

<sub>posted by Sider</sub>